src/coreclr/vm/peimagelayout.cpp

```diff
@@ -70,9 +87,13 @@ PEImageLayout* PEImageLayout::LoadConverted(PEImage* pOwner, bool disableMapping
         pFlat = (FlatImageLayout*)pOwner->GetFlatLayout();
         pFlat->AddRef();
     }
-    else if (pOwner->IsFile())
+    else
     {
-        pFlat = new FlatImageLayout(pOwner);
+        INT64 dataSize;
+        if (pOwner->IsFile() || pOwner->GetExternalData(&dataSize) != NULL)
```

Comment:
Can we use `AllowR2RForImage()` here or does the 0 size mean something special in this case?

Reply:
Composite R2R goes down this path - it was the only thing I found that did, so perhaps you are correct in that they are actually coupled and my broader interpretation of this case is not true.

Reply:
Okay - I think this should only be for R2R (composite, bundled). I switched to the helper function and added a comment and assert.